### DISCUSSION TITLE
Add new node to exist assignment with the least change

### DISCUSSION THREAD

**Discussion by Khixinhxan** [google/or-tools#2425] [Routing (and legacy CP) questions] [answered]
**What language and solver does this apply to?**
Python 3.7
or-tools 7.4.7247
Mac Os system
Routing

**Describe the problem you are trying to solve.**
Current, I want to find solution for add new node to exist assignment was solved by model
Example:
I have 40 nodes and **20 vehicles**, I using SolveWithParameters to solve.
First result I have **10 vehicles** as **10 routes**, cross 40 initial nodes.
Now, I have one new node, I want to add this new node to first result with the **least** change in the first result
But I re generate RoutingIndexManager, RoutingModel and use SolveWithParameters to solve then I will have new result, change more than 40 50 percents.

**Describe the solution you'd like**
I think two case to solve:
1. Add new node to new vehicle, I will have 11 vehicle as 11 routes.
2. I think search from first result, find the vehicle have free time, capacity of vehicle not full to add one new node, reset the ordered this route to optimize (min total time, min total distance, ...), so second result change only one vehicle.

**Describe alternatives you've considered**

**Additional context**

**Comment by Mizux** [comment, marked as answer]
* Add the new node in a disjunction so you can use the previous assignment as initial solution
see: https://gist.github.com/Mizux/0bd1daadc4e64130b62492c942f163d5

* force all previous node to the vehicle in the previous assignment so they won't be moved to another vehicle aka
```python
routing.VechicleVar(index_N).SetValue(vehicle_X)
```
* You can take a look at ApplyLock() method
https://github.com/google/or-tools/blob/fa84bc05e72641dddfbb98164d81b8bc9bef6ea5/ortools/constraint_solver/routing.h#L1043-L1061
cons: since the lock force the nextVar the new node won't/can't be inserted in the middle of the chain...
